src/components/Carousel/Carousel.tsx

```diff
@@ -1,176 +1,231 @@
-import { FC, ReactNode, useEffect, useRef, useState } from 'react';
-import style from './carousel.module.scss';
-import { Carousel } from 'antd';
-import SliderButtons from './SliderButtons';
-import { ResponsiveSettings } from '@/types/responsive';
-import type { CarouselRef } from 'antd/es/carousel';
-
-interface ICarouselsProps {
-  children: ReactNode;
+import React, { useState, ReactNode, useEffect } from 'react';
+import styles from './carousel.module.scss';
+import classNames from 'classnames';
+
+interface CustomCarouselProps {
   classname: string;
-  sliderClassName: string;
-  countSlideToShow: number;
-  responsive?: ResponsiveSettings[];
+  children: ReactNode[];
 }
 
-const Carousels: FC<ICarouselsProps> = ({
-  children,
+const CustomCarousel: React.FC<CustomCarouselProps> = ({
   classname,
-  sliderClassName,
-  countSlideToShow,
-  responsive,
+  children,
 }) => {
-  const ref = useRef<CarouselRef>(null);
-  const [prevClick, setPrevClick] = useState(false);
-  const [nextClick, setNextClick] = useState(false);
-  const [isFirstSlick, setIsFirstSlick] = useState(true);
-  const [isLastSlick, setIsLastSlick] = useState(false);
-  const containerRef = useRef<HTMLDivElement>(null);
-
-  const handlePrevClick = () => {
-    setPrevClick(true);
-    ref.current?.prev();
-    setTimeout(() => setPrevClick(false), 0); // Reset click state after 0s
-  };
+  const responsiveSettings = [
```

Comment:
This constant should be in a separate constants file or outside the component function. The first approach is more flexible